pkg/handler3/handler.go

```diff
@@ -96,10 +107,15 @@ func (o *OpenAPIService) getGroupBytes() ([]byte, error) {
 	}
 
 	sort.Strings(keys)
-	group := make(map[string][]string)
-	group["Paths"] = keys
-
-	j, err := json.Marshal(group)
+	discovery := &OpenAPIV3Discovery{Paths: make(map[string]string)}
+	for k, v := range o.v3Schema {
+		_, etag, err := v.jsonCache.Get()
+		if err != nil {
+			return nil, err
+		}
+		discovery.Paths[k] = "openapi/v3/" + k + "?hash=" + unquoteETag(etag)
```

Comment:
While consuming this API in a test I found it a bit unwieldy. I expected `discovery.Paths` to be a `groupVersion -> ETAG` map, rather than a `groupVersion -> path` map. I ended up using `url.Parse` to extract the `hash` query param from the path since the `client-go` discovery client for OpenAPI V3 [asks for the `hash` by itself](https://github.com/Jefftree/kubernetes/pull/3/files#diff-14f541e7356b7bc7f735b24db888b48b81b81e136b5af2925e912b95889dd010R476). 

Also it is worth pointing out that in the future more query params may be added to the OpenAPI endpoints, so it may become inconvenient to deal with these paths with an assumed URL structure.

Reply:
We may also in the future end up wanting to extend the discovery map to include additional metadata. Should the type of `discovery.Paths[k]` be a `map[string]interface{}` rather than simply a `string`? So that for example the response might be:

```json
{
    "paths": {
        "batch/v1": {
            "hash": <ETAG>
        }
    }
}
```

Reply:
I don't disagree with that, but it should be a URL rather than the hash so it's built automatically (clients shouldn't even have to know what to do with the hash, it should just contain the URL they need to download as a relative link).

Reply:
> While consuming this API in a test I found it a bit unwieldy. I expected discovery.Paths to be a groupVersion -> ETAG map, rather than a groupVersion -> path map. I ended up using url.Parse to extract the hash query param from the path since the client-go discovery client for OpenAPI V3 [asks for the hash by itself](https://github.com/Jefftree/kubernetes/pull/3/files#diff-14f541e7356b7bc7f735b24db888b48b81b81e136b5af2925e912b95889dd010R476).

Interesting, why didn't you just want to download the relative path given?

Reply:
Thanks for catching this inconsistency. 

>We may also in the future end up wanting to extend the discovery map to include additional metadata.

Good idea, I'll change the structure.

>  I ended up using url.Parse to extract the hash query param from the path since the client-go discovery client

Question for both of you: Would it make more sense for the client-go interface to directly consume the URL (eg: `OpenAPISchema(URL string)`) instead of the current `OpenAPISchema(gvPath, hash string)`. I agree we shouldn't assume an URL structure and going the route of parsing the URL may add more complexity in the future

Reply:
> I don't disagree with that, but it should be a URL rather than the hash so it's built automatically (clients shouldn't even have to know what to do with the hash, it should just contain the URL they need to download as a relative link).

In that case a URL could be included as one of the keys in the map I proposed

Reply:
Changed the Discovery schema format to

```
{
    "paths": {
        "apis/batch/v1": {
            "url":  "/openapi/v3/apis/batch/v1?hash=XXX"
        }
    }
}
```

Reply:
Yeah, I don't disagree that having an object rather than a direct mapping gives us more flexibility in the future, but we shouldn't need the hash per-se.